portfolio/src/test/java/com/google/sps/DataServletTest.java

```diff
@@ -56,127 +64,232 @@ public void tearDown() {
+    dataServelet.doPost(postRequest, postResponse);
     when(postRequest.getParameter("text-input")).thenReturn("Comment3");
-    new DataServlet().doPost(postRequest, postResponse);
+    dataServelet.doPost(postRequest, postResponse);
     
-    Assert.assertEquals(3, ds.prepare(new Query(COMMENT_TABLE_NAME)).countEntities(withLimit(10)));
+    assertThat(3).isEqualTo(ds.prepare(new Query(COMMENT_TABLE_NAME)).countEntities(withLimit(10)));
   }
 
   @Test 
-  public void testPOSTNull() throws Exception {
-    DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
-    HttpServletRequest postRequest = mock(HttpServletRequest.class);
-    HttpServletResponse postResponse = mock(HttpServletResponse.class);
-
-    Assert.assertEquals(0, ds.prepare(new Query(COMMENT_TABLE_NAME)).countEntities(withLimit(10)));
+  public void testPostNull() throws Exception {
+    assertThat(0).isEqualTo(ds.prepare(new Query(COMMENT_TABLE_NAME)).countEntities(withLimit(10)));
 
     when(postRequest.getParameter("text-input")).thenReturn(null);
-    new DataServlet().doPost(postRequest, postResponse);
+    dataServelet.doPost(postRequest, postResponse);
     
-    Assert.assertEquals(0, ds.prepare(new Query(COMMENT_TABLE_NAME)).countEntities(withLimit(10)));
+    assertThat(0).isEqualTo(ds.prepare(new Query(COMMENT_TABLE_NAME)).countEntities(withLimit(10)));
   }
 
   @Test
-  public void testGETSingleComment() throws Exception {
-    HttpServletRequest postRequest = mock(HttpServletRequest.class);
-    HttpServletResponse postResponse = mock(HttpServletResponse.class);
-    HttpServletRequest getRequest = mock(HttpServletRequest.class);
-    HttpServletResponse getResponse = mock(HttpServletResponse.class);
-
-    when(postRequest.getParameter("text-input")).thenReturn("Test Comment #1");
+  public void testGetSingleComment() throws Exception {
+    long TIMESTAMP_ONE = 0;
+    Entity comment = new Entity(COMMENT_TABLE_NAME);
+    comment.setProperty(COMMENT_COLUMN_NAME, "Test Comment #1");
+    comment.setProperty(TIMESTAMP_COLUMN_NAME, TIMESTAMP_ONE);
+    ds.put(comment);
 
     StringWriter stringWriter = new StringWriter();
     PrintWriter printWriter = new PrintWriter(stringWriter);
     when(getResponse.getWriter()).thenReturn(printWriter);
 
-    new DataServlet().doPost(postRequest, postResponse);
-    new DataServlet().doGet(getRequest, getResponse);
+    dataServelet.doPost(postRequest, postResponse);
+    dataServelet.doGet(getRequest, getResponse);
+
+    printWriter.flush(); //may not have flushed yet
+    String[] comments = convertStringToArray(stringWriter.toString());
+    assertThat(comments[0]).isEqualTo("Test Comment #1");
+  }
+  
+  @Test
+  public void testGetSomeComments() throws Exception {
+    int NUM_COMMENTS = 2;
+    long TIMESTAMP_ONE = 0;
+    Entity comment = new Entity(COMMENT_TABLE_NAME);
+    comment.setProperty(COMMENT_COLUMN_NAME, "Test Comment #1");
+    comment.setProperty(TIMESTAMP_COLUMN_NAME, TIMESTAMP_ONE);
```

Comment:
since this construct is used frequently it might be nice to abstract it the boilerplate behind a helper method and some constants:
```java

private static final Entity COMMENT_ENTITY_ONE = commentEntityOf("Comment 1", 1L);
private static final Entity COMMENT_ENTITY_TWO = commentEntityOf("Comment 2", 2L);

private static Entity commentEntityOf(String comment, long timestamp) {
  Entity entity = new Entity(COMMENT_TABLE_NAME);
  entity.setProperty(COMMENT_COLUMN_NAME, comment);
  entity.setProperty(TIMESTAMP_COLUMN_NAME, timestamp);
  return entity;
}
```

then you can reuse these comment entities in multiple tests

Reply:
thanks! updated!